extensions/2.0/Khronos/KHR_materials_transmission/README.md

```diff
@@ -0,0 +1,147 @@
+
+|   |Type|Description|Required|
+|---|----|-----------|--------|
+|**transmissionFactor** | `number` | The base percentage of light that is transmitted through the surface.| No, default:`1.0`|
+|**transmissionTexture** | [`textureInfo`] | A greyscale texture that defines the transmission percentage of the surface. This will be multiplied by transmissionFactor. | No |
+
+### transmissionFactor 
+
+The amount of light that is transmitted by the surface rather than diffusely re-emitted. This is a percentage of all the light that penetrates a surface (i.e. isn’t specularly reflected) rather than a percentage of the total light that hits a surface. A value of 1.0 means that 100% of the light that penetrates the surface is transmitted through. This is the default.
+
+<figure>
+  <img src="./figures/ConstantTransmission.png"/>
+<figcaption><em>Defaulting to 1.0 allows making a material transparent by simple defining the extension.</em></figcaption>
+</figure>
+
+### transmissionTexture 
+
+A greyscale texture that defines the amount of light that is transmitted by the surface rather than diffusely re-emitted. A value of 1.0 means that 100% of the light that penetrates the surface (i.e. isn’t specularly reflected) is transmitted through. The value is linear and is multiplied by the transmissionFactor to determine the total transmission value. Note that use of transmission values does not disallow usage of alpha coverage (via `baseColor.a`).
+
+<figure>
+  <img src="./figures/TransmissionTexture.png"/>
+<figcaption><em>Controlling transmission amount with transmissionTexture.</em></figcaption>
+</figure>
+
+## Implementing Transmission ##
+
+From the core [glTF BRDF](https://github.com/KhronosGroup/glTF/blob/master/specification/2.0/README.md#appendix-b-brdf-implementation), we have:
+
+*f* = *f*<sub>*diffuse*</sub> + *f*<sub>*specular*</sub>
+*f*<sub>*diffuse*</sub> = (1 - *F*) * *diffuse*
+*f*<sub>*specular*</sub> = *F* * *G* * *D* / (4 * dot(*N*, *L*) * dot(*N*, *V*))
+, where *F* is the Surface Reflection Ratio.
+
+We will now add an additional term for the transmitted light:
+
+*f* = *f*<sub>*diffuse*</sub> + *f*<sub>*specular*</sub> + *f*<sub>*transmission*</sub>
+
+*f*<sub>*transmission*</sub> = (1 - *F*) * *T* * *D<sub>T</sub>* * *baseColor*
+where *T* is the transmission percentage defined by this extension's `transmission` and `transmissionTexture` properties and *D<sub>T</sub>* is the distribution function for the transmitted light. The distribution function is the same Trowbridge-Reitz model used by specular reflection except sampled along the view vector rather than the reflection. The *baseColor* factor causes the transmitted light to be tinted by the surface.
```

Comment:
I personally would benefit from clarity here with regards to `Dt`.  This states that `Dt` is the Towbridge-Reitz distribution function, but sampled along the view direction rather than the reflected direction.  The microfacet distribution function, D, is calculated using dot(n, h).  Is it correct that this should instead use dot(n, -v) in this calculation for when we deal with transmission? Is it that simple?

Reply:
I think the `dot(n, h)` term shouldn't come into play at all since the transmitted light isn't being reflected, but I could be wrong. The incoming light direction should be sampled along the view direction because you're looking through the material and are seeing completely unrefracted light. The transmitted light is then scaled by the Fresnel to account for the portion of light reflected off the inside surface of the material (and, therefore, not reaching your eye).
Not sure at all if this is correct so it would be good to have others (like @proog128) look at this.

Reply:
To my understanding, the dot product is relevant as we still evaluate the microfacet distribution. There's a brief section on the topic in the DSPBR spec https://bit.ly/2NpmJVn (eq. 5 and fig. 4). We just use a modified half-vector, which uses the flipped light direction.

Reply:
Yes, you're right. You'd flip it across the plane defined by the surface normal and then it would be centered around the same direction as the view.

Reply:
In this sentence you mention Trowbridge-Reitz by name, but we only suggest that model in Appendix B as non-normative.  Should it be mentioned here?  I would consider rewriting this as:
```suggestion
where *T* is the transmission percentage defined by this extension's `transmission` and `transmissionTexture` properties and *D<sub>T</sub>* and *G<sub>T</sub>* are the microfacet distribution and shadowing functions for the transmitted light. These *D<sub>T</sub>* and *G<sub>T</sub>* functions are the same models used by specular reflection, except sampled along the view vector rather than the reflection. The *baseColor* factor causes the transmitted light to be tinted by the surface.
```